src/Nox.Ui.Blazor.Lib/Components/NoxTypes/ViewTemperature.razor.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.AspNetCore.Components;
+using Nox.Types;
+using Nox.Ui.Blazor.Lib.Models;
+
+namespace Nox.Ui.Blazor.Lib.Components.NoxTypes;
+
+public partial class ViewTemperature : ComponentBase
+{
+    #region Declarations
+
+    [Parameter]
+    public decimal? Temperature { get; set; }
+
+    [Parameter]
+    public TemperatureUnit TemperatureUnit { get; set; } = TemperatureUnit.Celsius;
+
+    [Parameter]
+    public string Format { get; set; } = "#.##";
+
+    public string DisplayTemperature
+    {
+
+        get
+        {
+            if (Temperature.HasValue)
+            {
+                return Temperature.Value.ToString(Format) + DisplayTemperatureUnit();
+            }
+
+            return String.Empty;
+        }
+    }
+
+    private string DisplayTemperatureUnit()
+    {
+        if (TemperatureUnit == TemperatureUnit.Fahrenheit)
+        {
+            return " " + TemperatureUnit.Fahrenheit.Symbol;
```

Comment:
I would suggest this "Empty Space" prefix to be added on the DisplayTemperature fproperty, since this one should return only the units... this can be difficult to maintain in the future

Reply:
amended in latest commit - plus Area, Length, Distance as well